File: models/Fuel.py
from models.Serializer import Serializer


class Fuel(Serializer):
    fields = [
        "id",
        "vehicle",
        "odometer",
        "tripometer",
        "oil_level",
        "gallons",
        "dollars",
        "date"
    ]

    def __init__(
            self,
            id=None,
            vehicle=None,
            odometer=None,
            tripometer=None,
            oil_level=None,
            gallons=None,
            dollars=None,
            date=None
    ):
        self.id = id
        self.vehicle = vehicle
        self.tripometer = tripometer
        self.odometer = odometer
        self.oil_level = oil_level
        self.gallons = gallons
        self.dollars = dollars
        self.date = date

File: src/cp.models/fuel.js
const {Model, DataTypes} = require('sequelize')
const sequelize = require('./connection')

class Fuel extends Model {}

Fuel.init(
  {
    id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      autoIncrement: true,
      primaryKey: true,
    },
    vehicle: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: 'vehicle',
      onDelete: 'CASCADE',
    },
    tripometer: {
      type: 'Decimal(8,2)',
    },
    odometer: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    oil_level: {
      type: 'Decimal(8,2)',
    },
    gallons: {
      type: 'Decimal(8,2)',
    },
    dollars: {
      type: 'Decimal(8,2)',
    },
    date: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: DataTypes.NOW,
    },
  },
  {sequelize, modelName: 'fuel', timestamps: false},
)

module.exports = Fuel
